engine/src/graphics/device.cpp

```diff
@@ -132,13 +135,64 @@ namespace gen
 			queueCreateInfos.push_back(queueCreateInfo);
 		}
 
-		vk::DeviceCreateInfo const createInfo({}, static_cast<u32>(queueCreateInfos.size()), queueCreateInfos.data());
+		vk::DeviceCreateInfo const createInfo({}, static_cast<u32>(queueCreateInfos.size()), queueCreateInfos.data(),
+											  0,	   // ignored by spec
+											  nullptr, // ignored by spec
+											  static_cast<u32>(deviceExtensions.size()), deviceExtensions.data(),
+											  nullptr // to be used later
+		);
 
 		m_device = m_gpu.physicalDevice.createDeviceUnique(createInfo);
 
+		// display all device extensions using m_device
+		auto availableExtensions = m_gpu.physicalDevice.enumerateDeviceExtensionProperties();
+		auto aExt				 = std::stringstream{"vulkan"};
+		for (std::size_t i = 0; i < availableExtensions.size(); i++)
+		{
+			aExt << "\tExtension [" << i << "]"
+				 << " | Name: " << availableExtensions[i].extensionName << " | Spec Ver: " << availableExtensions[i].specVersion << "\n";
+		}
+		m_logger.debug("Found Device Extensions: \n{}\n", aExt.str());
+
 		m_graphicsQueue = m_device->getQueue(m_gpu.queueFamily, 0);
 	}
 
+	void GraphicsDevice::createSwapChain(const Window & window)
+	{
+		SwapChainSupportDetails const swapChainSupport = querySwapChainSupport(m_gpu.physicalDevice, m_surface.get());
+		assert(!swapChainSupport.formats.empty());
+
+		vk::SurfaceFormatKHR const surfaceFormat = chooseSwapSurfaceFormat(swapChainSupport.formats);
+
+		vk::Extent2D swapchainExtent{};
+		int width{}, height{}; // NOLINT
+		glfwGetFramebufferSize(window.getHandle(), &width, &height);
```

Comment:
This function has been removed in favor of using glfw directly. I personally felt it was outside of the scope of allowing for the window class to be able to return the frame buffer since this in itself is pretty tied to vulkan. So I felt it made more sense to just call directly from glfw as this will always be the most up to date buffer.